metecho/api/serializers.py

```diff
@@ -364,23 +364,55 @@ def update(self, instance, validated_data):
         if instance.assigned_dev != validated_data["assigned_dev"]:
             if validated_data.get("should_alert_dev"):
                 self.try_send_assignment_emails(instance, "dev", validated_data, user)
-            orgs = instance.scratchorg_set.active().filter(
-                org_type=SCRATCH_ORG_TYPES.Dev
-            )
+            # We want to consider soft-deleted orgs, too:
+            orgs = instance.scratchorg_set.all().filter(org_type=SCRATCH_ORG_TYPES.Dev)
```

Comment:
@wlonk Is this iterating order too arbitrary? It seems possible that multiple orgs (some soft-deleted) could be potentially re-assigned... If there is an active org already in this task-role, we should definitely be checking that first -- then looking through soft-deleted orgs, probably in `-created_at` order?